lib/datadog/core/remote/component.rb

```diff
@@ -0,0 +1,42 @@
+# frozen_string_literal: true
+
+require_relative 'worker'
+require_relative 'client'
+require_relative '../transport/http'
+require_relative '../remote'
+
+module Datadog
+  module Core
+    module Remote
+      class Component
+        attr_reader :client
+
+        def initialize(settings, agent_settings)
+          transport_options = {}
+          transport_options[:agent_settings] = agent_settings if agent_settings
+
+          @transport_v7 = Datadog::Core::Transport::HTTP.v7(**transport_options.dup)
+
+          @client = Client.new(@transport_v7)
+          @worker = Worker.new(interval: 1) { @client.sync }
```

Comment:
Will the interval be something that could be configured via configuration?

Reply:
At minimum, extract it to a constant locally, something like `DEFAULT_INTERVAL =  1`.

Reply:
(Also, consider including the unit -- I'm guessing seconds?)